src/lib/theme/theme.service.ts

```diff
@@ -19,23 +19,37 @@ export interface IVantageThemeMap {
 export class VantageThemeService {
   private _renderer2: Renderer2;
 
-  private readonly _activeThemeSubject: BehaviorSubject<VantageTheme> = new BehaviorSubject<VantageTheme>(
-    <VantageTheme>localStorage.getItem(THEME_LOCAL_STORAGE_KEY),
-  );
+  private readonly _activeThemeSubject: BehaviorSubject<VantageTheme>;
 
-  public readonly activeTheme$: Observable<VantageTheme> = this._activeThemeSubject.asObservable();
-  public readonly darkTheme$: Observable<boolean> = this._activeThemeSubject
-    .asObservable()
-    .pipe(map((theme: VantageTheme) => theme === VantageTheme.DARK));
-  public readonly lightTheme$: Observable<boolean> = this._activeThemeSubject
-    .asObservable()
-    .pipe(map((theme: VantageTheme) => theme === VantageTheme.LIGHT));
+  public activeTheme$: Observable<VantageTheme>;
+  public darkTheme$: Observable<boolean>;
+  public lightTheme$: Observable<boolean>;
 
   constructor(private rendererFactory: RendererFactory2, @Inject(DOCUMENT) private _document: any) {
+    const initialValue: VantageTheme =
+      <VantageTheme>localStorage.getItem(THEME_LOCAL_STORAGE_KEY) || this.checkOSPreference();
+
     this._renderer2 = rendererFactory.createRenderer(undefined, undefined);
+    this._activeThemeSubject = new BehaviorSubject<VantageTheme>(initialValue);
+
+    this.activeTheme$ = this._activeThemeSubject.asObservable();
+    this.darkTheme$ = this._activeThemeSubject
+      .asObservable()
+      .pipe(map((theme: VantageTheme) => theme === VantageTheme.DARK));
+    this.lightTheme$ = this._activeThemeSubject
+      .asObservable()
+      .pipe(map((theme: VantageTheme) => theme === VantageTheme.LIGHT));
+
+    // apply initial theme
+    this.applyTheme(initialValue, false);
```

Comment:
So why don't we want to save the theme in this case. I wanted the local storage to always be the source of truth.

Reply:
right but now that we are trying to have the OS pref be the first source of truth we can't overwrite in local storage till a user explicitly switches their preference by toggling